app/domain/authentication/authenticate.rb

```diff
@@ -22,8 +22,8 @@ module Authentication
     def call
       validate_authenticator_exists
       validate_security
```

Comment:
Is there a check here that does not relate to security?
As @shaharglazner pointed out. this is not clear

Reply:
what `validate_security` does is validate that the authenticator is enables, it has a defined webservice, the user exists and that it has authenticate permissions on the webservice.

Is it still unclear?

Reply:
`validate_authenticator_exists` validates that the authenticator is implemented in Conjur, not that it is enabled in the env.

I agree that `validate_security` is quite vague and we should consider making it clearer.

Reply:
`validate_security` validates several check. we can call each one explicitly instead of stacking them in that class (i.e move all the validations preformed in `validate_security` to the calling class) but i'd like to get some thoughts on this from @micahlee & @jonahx .

Reply:
for visualization, the `authenticate` call method now looks like this:
```
      validate_authenticator_exists
      validate_security
      validate_origin
      validate_credentials
      audit_success
      new_token
```

If we move the validations from `validate_security` to the calling class, the call method will look like this:
```
      validate_authenticator_exists
      validate_webservice_is_whitelisted
      validate_user_has_access_to_webservice
      validate_origin
      validate_credentials
      audit_success
      new_token
```

@Tovli @shaharglazner @micahlee @jonahx wdyt?

Reply:
I think this is clearer than just `validate_security`. I'm also a fan of smaller PRs, so I suggest making this a follow up rather than adding all of the improvements to a single PR. It will make it much easier to review more quickly.